### DISCUSSION TITLE
R, Rstudio and virtual environment

### DISCUSSION THREAD

**Discussion by theb8** [ateucher/pkg-dev-dfo-2024-11#34] [General]
With python it is common to work with a virtual environment (venv) for specific projects, is it a viable option in R ?

In my experience R and Rstudio were unstable when installed in a venv, but I've seen that there are different packages out there like packrat and renv. 

In a developper context it could be useful for reproducibility and debugging.

Do you have an opinion on wether it is beneficial to integrate a venv approach to the workflow ?

Thank you

**Comment by ateucher** [comment]
Hi @theb8 - good question. You're right, venv doesn't play well with R. renv is great for _projects_ where having stable and consistent package versions is important for reproducibility. It doesn't work great for developing packages, so I would avoid it in that context. You can't ship an renv environment with your package. The way you ensure that users of your package have dependency versions that are appropriate for your package is by specifying a minimum version in the DESCRIPTION file.
